### PR TITLE
Remove crypto

### PR DESCRIPTION
**Pull request by Patater**
Remove all cryptography files as they are sourced from Mbed Crypto. This helps with maintenance of the library by avoiding duplicate files between Mbed TLS and Mbed Crypto.

## Status
**READY**

## Requires Backporting
NO

## Migrations
NO

## Additional comments
Any additional information that could be of interest

## Todos
- [ ] Tests
- [ ] Documentation
- ~[ ] Changelog updated~
- ~[ ] Backported~

### CODE REVIEW

**Comment by yanesca** [comment]
Could you please fix the CI failures before we start the reviews?

**Comment by Patater** [comment]
Added a commit to update the crypto submodule to a version that includes the latest crypto and configuration and tooling changes, to fix `undefined reference to `mbedtls_param_failed'
` errors.

**Comment by Patater** [comment]
Rebased to use a commit on the crypto/development branch in commit "Update crypto to a repo with latest crypto"

File: include/mbedtls/config.h [hunk 1]
```diff
@@ -295,7 +295,7 @@
  *   the function mbedtls_param_failed() in your application.
  *   See `platform_util.h` for its prototype.
  * - If you enable the macro #MBEDTLS_CHECK_PARAMS_ASSERT, then the
- *   library defines #MBEDTLS_PARAM_FAILED(\c cond) to be `assert(cond)`.
+ *   library defines MBEDTLS_PARAM_FAILED(\c cond) to be `assert(cond)`.
```
[the remark this answers is not in the document]
Comment (Patater) [reply include/mbedtls/config.h]: https://github.com/ARMmbed/mbed-crypto/issues/195

**Comment by Patater** [comment]
Rebased on latest development to resolve conflicts. Conflicts resolved by telling git to still delete the files we want to delete, even after the changes they received in upstream. No other changes.

**Comment by yanesca** [comment]
If there were changes upstream, then I think we should do a merge of Mbed TLS into Mbed Crypto, before this PR goes in.

**Comment by Patater** [comment]
> If there were changes upstream, then I think we should do a merge of Mbed TLS into Mbed Crypto, before this PR goes in.

I've raised https://github.com/ARMmbed/mbed-crypto/pull/223 for this.

**Comment by Patater** [comment]
Rebased to update to latest crypto, which contains latest changes from Mbed TLS.

**Comment by Patater** [comment]
Testing went well except for Mbed OS testing. Mbed OS PR https://github.com/ARMmbed/mbed-os/pull/11239 will fix.

**Comment by Patater** [comment]
Testing with the Mbed OS PR at https://jenkins-internal.mbed.com/blue/organizations/jenkins/mbedtls-release-new/detail/mbedtls-release-new/425/pipeline/263/

**Pull request merged**